File: dataset/human/python/file_5033.py
import scala.collection.mutable

object Main extends App {
  val sc = new java.util.Scanner(System.in)
  val n, m = sc.nextInt

  val fs = mutable.Set.empty[Int]
  val es = mutable.Set.empty[Int]

  (1 to m).foreach { _ =>
    val ai, bi = sc.nextInt
    if (ai == 1) fs.add(bi)
    if (bi == n) es.add(ai)
  }
  println(if ((fs & es).nonEmpty) "POSSIBLE" else "IMPOSSIBLE")
}

File: dataset/human/java/file_5033.java
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        StringBuffer s = new StringBuffer(sc.next());
        int q = sc.nextInt();
        int pn = 0;
        StringBuffer plus = new StringBuffer();
        StringBuffer minus = new StringBuffer();
        for (int i=0;i<q;i++){
            if (sc.nextInt()==1){
                pn++;
            }else {
                if ((sc.nextInt()-1+pn)%2==0){
                    plus.append(sc.next());
                }else {
                    minus.append(sc.next());
                }
            }
        }
        if (pn%2==1){
            plus.reverse();
            s.append(minus);
            s.insert(0,plus);
        }else {
            s.reverse();
            minus.reverse();
            s.append(plus);
            s.insert(0,minus);
        }
        s.reverse();
        System.out.println(s);
    }
}
